src/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,48 @@
+import { useMutation } from "@apollo/client";
+import { useState, createContext, useContext, useEffect, FC } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { LOGIN, Login, User } from "../components/pages/LoginPage/interface";
+import { AuthContextType, AuthProps } from "./interface";
+
+const AuthContext = createContext<AuthContextType>({} as AuthContextType);
+
+export const AuthProvider: FC<AuthProps> = ({ children }) => {
+  const [user, setUser] = useState<User>();
+
+  const navigate = useNavigate();
+
+  const userData = localStorage.getItem("user");
+  const parsedUserData = userData && JSON.parse(userData);
+
+  useEffect(() => {
+    if (!!parsedUserData == true) {
+      setUser(parsedUserData);
+    }
+  }, []);
```

Comment:
What if user is removed from local storage when the app is already mounted? There is nothing listening to this change - we can sort it out in separate PR though. :)